### PR TITLE
conanized zlib-ng 2.0.0

### PR DESCRIPTION
**Pull request by tomas-krupa**
Specify library name and version:  **zlib-ng/2.0.0**

zlib data compression library for the next generation systems

Author:  Hans Kristian Rosbach aka Dead2 
Homepage: https://github.com/zlib-ng/zlib-ng

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zlib-ng/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    description = "zlib data compression library for the next generation systems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage="https://github.com/Dead2/zlib-ng/"
+    license="https://github.com/Dead2/zlib-ng/blob/ba7f0eb6e294306ac6d771216ea4442f2ea2d830/LICENSE.md"
+    topics = ("conan", "zlib", "compression")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False]}
+    default_options = "shared=False"
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:16]: Add fPIC + use dicts
```suggestion
    options = {"shared": [True, False], "fPIC": [True, False]}
    default_options = ["shared": False, "fPIC": True}
```

File: recipes/zlib-ng/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:30]: Handle fPIC
```suggestion
        return "build_subfolder"

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

File: recipes/zlib-ng/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    self.cpp_info.libs = ['zlibd']
+                else:
+                    self.cpp_info.libs = ['zlib']
+            else:
+                if self.settings.build_type == "Debug":
+                    self.cpp_info.libs = ['zlibstaticd']
+                else:
+                    self.cpp_info.libs = ['zlibstatic']
+        else:
+            self.cpp_info.libs = ['z-ng']
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:72]: double quotes
```suggestion
        if self.settings.os == "Windows":
            if self.options.shared:
                if self.settings.build_type == "Debug":
                    self.cpp_info.libs = ["zlibd"]
                else:
                    self.cpp_info.libs = ["zlib"]
            else:
                if self.settings.build_type == "Debug":
                    self.cpp_info.libs = ["zlibstaticd"]
                else:
                    self.cpp_info.libs = ["zlibstatic"]
        else:
            self.cpp_info.libs = ["z-ng"]
```

File: recipes/zlib-ng/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        if self.options.shared:
+            self.copy(pattern="*.dylib*", dst="lib", src=self._build_subfolder, keep_path=False, symlinks=True)
+            self.copy(pattern="*.so*", dst="lib", src=self._build_subfolder, keep_path=False, symlinks=True)
+            self.copy(pattern="*.dll", dst="bin", src=self._build_subfolder, keep_path=False)
+            self.copy(pattern="*.dll.a", dst="lib", src=self._build_subfolder, keep_path=False)
+        else:
+            self.copy(pattern="*.a", dst="lib", src=self._build_subfolder, keep_path=False)
+        self.copy(pattern="*.lib", dst="lib", src=self._build_subfolder, keep_path=False)
+
+        self._rename_libraries()
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:114]: please insert final newline
```suggestion
        self._rename_libraries()
```

File: recipes/zlib-ng/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package LANGUAGES C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+find_package(ZLIB)
```
Comment (madebr) [inline recipes/zlib-ng/all/test_package/CMakeLists.txt:7]:
```suggestion
find_package(ZLIB REQUIRED)
```

File: recipes/zlib-ng/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        cmake.build()
+
+    def test(self):
+        bin_path = os.path.join("bin", "test_package")
+        buffer = StringIO()
+        try:
+           self.run(bin_path, run_environment=True, output=buffer)
+           self.output.success("enough.c test passed")
+        except:
+            self.output.error(buffer.getvalue())
```
Comment (madebr) [inline recipes/zlib-ng/all/test_package/conanfile.py:21]: There's no need to parse the output. Just check the return code.
```suggestion
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

File: recipes/zlib-ng/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from conans import ConanFile, CMake
+from io import StringIO
+import os
```
Comment (madebr) [inline recipes/zlib-ng/all/test_package/conanfile.py:3]:
```suggestion
from conans import ConanFile, CMake, tools
import os
```

File: recipes/zlib-ng/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
+from conans import ConanFile, CMake
+from conans.tools import download, unzip, replace_in_file, remove_files_by_mask, get
+import os
+
+class ZlibNgConan(ConanFile):
+    name = "zlib-ng"
+    description = "zlib data compression library for the next generation systems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage="https://github.com/Dead2/zlib-ng/"
+    license="https://github.com/Dead2/zlib-ng/blob/ba7f0eb6e294306ac6d771216ea4442f2ea2d830/LICENSE.md"
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:10]: If I'm not mistaken, this is the ZLIB license.
See https://spdx.org/licenses/Zlib.html
```suggestion
    license ="Zlib"
```

File: recipes/zlib-ng/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
+from conans import ConanFile, CMake
+from conans.tools import download, unzip, replace_in_file, remove_files_by_mask, get
+import os
+
+class ZlibNgConan(ConanFile):
+    name = "zlib-ng"
+    description = "zlib data compression library for the next generation systems"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage="https://github.com/Dead2/zlib-ng/"
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:9]: Spaces around `=`
```suggestion
    homepage = "https://github.com/Dead2/zlib-ng/"
```

File: recipes/zlib-ng/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+        # Copy license
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+
+        # Copy headers
+        for header in ["*zlib.h", "*zconf.h"]:
+            self.copy(pattern=header, dst="include", src=self._source_subfolder, keep_path=False)
+            self.copy(pattern=header, dst="include", src=self._build_subfolder, keep_path=False)
+
+        # Copying static and dynamic libs
+        #build_dir = os.path.join(self._source_subfolder, "_build")
```
Comment (madebr) [inline recipes/zlib-ng/all/conanfile.py:104]: Remove dead code.
```suggestion
```

**Review by madebr — COMMENTED** [review]
Thanks for the pr.
I've got some small remarks.

File: recipes/zlib-ng/all/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/zlib-ng/all/conanfile.py:72]: I know that libraries are renamed in zlib recipe, but is it really something we should do?

File: recipes/zlib-ng/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+            else:
+                if self.settings.compiler == "Visual Studio":
+                    current_lib = os.path.join(lib_path, "zlibstatic%s.lib" % suffix)
+                    os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
+                elif self.settings.compiler == "gcc":
+                    current_lib = os.path.join(lib_path, "libzlibstatic.a")
+                    os.rename(current_lib, os.path.join(lib_path, "libzlib.a"))
+                elif self.settings.compiler == "clang":
+                    current_lib = os.path.join(lib_path, "zlibstatic.lib")
+                    os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
```
Comment (SpaceIm) [inline recipes/zlib-ng/all/conanfile.py:92]: libraries are renamed here, but in the meantime you define lib names in package_info() like if they were not renamed.

File: recipes/zlib-ng/all/test_package/conanfile.py (hunk 6 above)
Comment (tomas-krupa) [reply recipes/zlib-ng/all/test_package/conanfile.py:21]: not parsing, just hiding excessive output of the test

**Comment by uilianries** [comment]
@tomas-krupa I think you used a second email for these commits. CLA is asking for tkrupa user too.

https://github.com/conan-io/conan-center-index/pull/5200#issuecomment-818038711

**Comment by tomas-krupa** [comment]
sorry for my CLA issues, please proceed to  PR #5264

**Pull request closed without merging**